src/xdg-desktop-portal.c

```diff
@@ -355,6 +356,29 @@ on_bus_acquired (GDBusConnection *connection,
 
   g_debug ("providing portal %s", g_dbus_interface_skeleton_get_info (skeleton)->name);
   }
+
+  {
+  GDBusInterfaceSkeleton *skeleton = proxy_resolver_create (connection);
+  if (skeleton != NULL)
+    {
+      g_dbus_interface_skeleton_set_flags (skeleton,
+                                           G_DBUS_INTERFACE_SKELETON_FLAGS_HANDLE_METHOD_INVOCATIONS_IN_THREAD);
```

Comment:
Actually, the libproxy implementation just uses a thread to make it async, so maybe you should instead just use the sync call and RUN_IN_THREAD.